mewarpx/mewarpx/emission.py

```diff
@@ -0,0 +1,867 @@
+                half-step using the E-field. Aligns position and velocities
+                correctly for the leapfrog algorithm.
+        Returns:
+            particle_dict (dict): Contains lists, each with length equal to the
+            number of particles:
+                - ``x``, ``y``, and ``z`` contain initial positions
+                - ``vx``, ``vy``, and ``vz`` contain initial velocities
+                - ``E_total`` contains initial energy of each particle, kinetic
+                  & potential.
+                - ``w`` contains particle weights.
+        """
+        if rseed is not None:
+            nprstate = np.random.get_state()
+            np.random.seed(rseed)
+            rseedxv = np.random.randint(1000000000)
+            rseedt = np.random.randint(1000000000)
+        else:
+            rseedxv = None
+            rseedt = None
+
+        x, y, z, vx, vy, vz = self._get_xv_coords(
+            npart=npart, m=m, rseed=rseedxv
+        )
+        particle_dict = self._gen_particle_dict(
+            x=x, y=y, z=z, vx=vx, vy=vy, vz=vz, w=w
+        )
+
+        if self._wfnlist is not None:
+            for wfn in self._wfnlist:
+                particle_dict['w'] = wfn(particle_dict)
+
+        # if (self.use_Schottky and abs(q + e)/e < 1e-6
+        #         and abs(m - m_e)/m_e < 1e-6):
+        #     particle_dict['w'] = self.apply_Schottky_weights(particle_dict)
+
+        particle_dict['E_total'] = self._get_E_total(
+            vx=particle_dict['vx'],
+            vy=particle_dict['vy'],
+            vz=particle_dict['vz'],
+            q=q, m=m, w=particle_dict['w']
+        )
+
+        # After E_total has been computed, we advance particles as needed.
+        if randomdt:
+            self.particle_helper.advance_random_deltat(
+                particle_dict['x'], particle_dict['y'], particle_dict['z'],
+                particle_dict['vx'], particle_dict['vy'], particle_dict['vz'],
+                q=q, m=m, rseed=rseedt
+            )
+
+        if velhalfstep:
+            self.particle_helper.push_v_minus_halfstep(
+                particle_dict['x'], particle_dict['y'], particle_dict['z'],
+                particle_dict['vx'], particle_dict['vy'], particle_dict['vz'],
+                q=q, m=m
+            )
+
+        # TODO: Figure out wrapping after the randomdt/velhalfstep. We should
+        # check periodic BCs manually and handle dimensions though, so it's a
+        # fair bit of work.
+        # # Manually warp around periodic BCs; otherwise warp will autoscrape
+        # # particles outside
+        # x_coords = np.stack(
+        #     (particle_dict['x'], particle_dict['y'], particle_dict['z']),
+        #     axis=-1
+        # )
+        # x_coords = util.wrap_positions(x_coords)
+        # particle_dict['x'] = x_coords[:, 0]
+        # particle_dict['y'] = x_coords[:, 1]
+        # particle_dict['z'] = x_coords[:, 2]
+
```

Comment:
Based on @KZhu-ME 's statement that WarpX automatically wraps positions this whole block can be removed.